src/components/input/input.scss

```diff
@@ -138,6 +138,10 @@ md-input-container {
       outline: none;
       box-shadow: none;
     }
+
+    &.md-no-flex {
+      flex: none !important;
```

Comment:
I don't really love this solution, as md-no-flex is an "utility" class. But I had no success defining temporarily the 'flex: none' on the element directly (where it would take precedence over all the other rules).

Do you have any suggestions?